ui-v2/app/adapters/node.js

```diff
@@ -17,83 +17,21 @@ export default Adapter.extend({
-  },
-  isQueryLeader: function(url, method) {
-    return url.pathname === this.parseURL(this.urlForQueryLeader({})).pathname;
-  },
-  queryLeader: function(store, modelClass, id, snapshot) {
-    const params = {
-      store: store,
-      type: modelClass,
-      id: id,
-      snapshot: snapshot,
-      requestType: 'queryLeader',
-    };
-    // _requestFor is private... but these methods aren't, until they disappear..
-    const request = {
-      method: this.methodForRequest(params),
-      url: this.urlForRequest(params),
-      headers: this.headersForRequest(params),
-      data: this.dataForRequest(params),
-    };
-    // TODO: private..
-    return this._makeRequest(request);
-  },
-  handleBatchResponse: function(url, response, primary, slug) {
-    const dc = url.searchParams.get(API_DATACENTER_KEY) || '';
-    return response.map((item, i, arr) => {
-      // this can go in the serializer
-      item = fillSlug(item);
-      // this could be replaced by handleSingleResponse
-      // maybe perf test first although even polyfilled searchParams should be super fast
-      return {
-        ...item,
-        ...{
-          [DATACENTER_KEY]: dc,
-          [PRIMARY_KEY]: this.uidForURL(url, item[SLUG_KEY]),
-        },
-      };
-    });
+  requestForQueryLeader: function(request, { dc }) {
+    return request`
+      GET /v1/status/leader?${{ dc }}
+    `;
   },
-  handleResponse: function(status, headers, payload, requestData) {
-    let response = payload;
-    const method = requestData.method;
-    if (status === HTTP_OK) {
-      const url = this.parseURL(requestData.url);
-      let temp, port, address;
-      switch (true) {
-        case this.isQueryLeader(url, method):
-          // This response is just an ip:port like `"10.0.0.1:8000"`
-          // split it and make it look like a `C`onsul.`R`esponse
-          // popping off the end for ports should cover us for IPv6 addresses
-          // as we should always get a `address:port` or `[a:dd:re:ss]:port` combo
-          temp = response.split(':');
-          port = temp.pop();
-          address = temp.join(':');
-          response = {
-            Address: address,
-            Port: port,
-          };
-          break;
-        case this.isQueryRecord(url, method):
-          response = this.handleSingleResponse(url, fillSlug(response), PRIMARY_KEY, SLUG_KEY);
-          break;
-        default:
-          response = this.handleBatchResponse(url, response, PRIMARY_KEY, SLUG_KEY);
-      }
-    }
-    return this._super(status, headers, response, requestData);
+  queryLeader: function(store, type, id, snapshot) {
+    return this.request(
+      function(adapter, request, serialized, unserialized) {
+        return adapter.requestForQueryLeader(request, serialized, unserialized);
+      },
+      function(serializer, response, serialized, unserialized) {
+        return serializer.respondForQueryLeader(response, serialized, unserialized);
```

Comment:
Hmmm yeah, thinking back I wanted it to match up with `requestFor...`, but I suppose I was thinking of it as a verb rather than a noun - I kinda tried to make this look like the old `urlFor...` things which uses nouns, so maybe these should be thought of as nouns also as you suggest here. Let me take another look at it, the reasons why (if there were any) might have got lost in the refactor.

Reply:
Not looked in detail yet buck, but just realized something else here which might be more 'correct' sounding. `request` and `response` are actually functions, that when called 'do' the request things, or the response things. So I'm considering the option of changing the name of the argument to `respond` here. As its a function, a verb usually fits better. Just wanted to P.S. that on here, I'm gonna take a closer look now.

Reply:
Hey, so, I went with changing the arg to be called `respond`, which as its a function looks more 'function-y' - lemme know what you think.

> Also, the function doesn’t look like it uses the third parameter, is there a reason to include it in this call?

I generally keep unused arguments in there, mainly for basic documentation (somebody else might come along and see they have `unserialized` at their disposal within the function's scope even though it wasn't used previously) and for consistency.

Ununsed arguments will get uglified out in the build also